bpf/ingress_node_firewall_kernel.c

```diff
@@ -51,14 +51,24 @@ struct {
 
 __attribute__((__always_inline__)) static inline int
 ip_extract_l4Info(void *dataStart, void *dataEnd, __u8 *proto, __u16 *dstPort,
-                  __u8 *icmpType, __u8 *icmpCode) {
-    struct iphdr *iph = dataStart;
-    dataStart += sizeof(struct iphdr);
-    if (unlikely(dataStart > dataEnd)) {
-        return -1;
-    }
-	*proto = iph->protocol;
-	switch (iph->protocol) {
+                  __u8 *icmpType, __u8 *icmpCode, __u8 is_v4) {
+
+    if (likely(is_v4)) {
+        struct iphdr *iph = dataStart;
+        dataStart += sizeof(struct iphdr);
+        if (unlikely(dataStart > dataEnd)) {
+            return -1;
+        }
+		*proto = iph->protocol;
+	} else {
```

Comment:
Pull this "else" back 4 spaces. Hard to read.

Reply:
This may merge before you fix this because I put lgtm on it. You can follow up with a PR to format it.

Reply:
oh sorry will fmt it Thanks for the review